lib/curl_ntlm_core.c

```diff
@@ -498,8 +498,10 @@ CURLcode Curl_ntlm_core_mk_nt_hash(struct Curl_easy *data,
    * network encoding not the host encoding.
    */
   result = Curl_convert_to_network(data, (char *)pw, len * 2);
-  if(result)
+  if(result) {
+    free(pw);
     return result;
+  }
```

Comment:
To avoid two calls to free() for the same pointer, I propose this diff instead:
~~~diff
diff --git a/lib/curl_ntlm_core.c b/lib/curl_ntlm_core.c
index 89d4ec872..ef9b7480b 100644
--- a/lib/curl_ntlm_core.c
+++ b/lib/curl_ntlm_core.c
@@ -496,21 +496,18 @@ CURLcode Curl_ntlm_core_mk_nt_hash(struct Curl_easy *data,
   /*
    * The NT hashed password needs to be created using the password in the
    * network encoding not the host encoding.
    */
   result = Curl_convert_to_network(data, (char *)pw, len * 2);
-  if(result)
-    return result;
-
-  /* Create NT hashed password. */
-  Curl_md4it(ntbuffer, pw, 2 * len);
-
-  memset(ntbuffer + 16, 0, 21 - 16);
-
+  if(!result) {
+    /* Create NT hashed password. */
+    Curl_md4it(ntbuffer, pw, 2 * len);
+    memset(ntbuffer + 16, 0, 21 - 16);
+  }
   free(pw);
 
-  return CURLE_OK;
+  return result;
 }
 
 #if defined(USE_NTLM_V2) && !defined(USE_WINDOWS_SSPI)
 
 /* Timestamp in tenths of a microsecond since January 1, 1601 00:00:00 UTC. */

Reply:
Looks good.